starboard/common/time.cc

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "starboard/common/time.h"
+
+#if SB_API_VERSION >= 16
+
+#include <sys/time.h>
+#include <time.h>
+
+#include "starboard/common/log.h"
+
+#else  // SB_API_VERSION >= 16
+
+#include "starboard/time.h"
+
+#endif  // SB_API_VERSION >= 16
+
+namespace starboard {
+
+int64_t CurrentMonotonicTime() {
```

Comment:
What's the reason we define starboard::CurrentMonotonicTime instead of using clock_gettime in code directly with musl dealing with any necessary redirection?

Reply:
That makes sense—my preference would be that this live somewhere other than //starboard/common, as seeing references to the starboard namespace in cobalt code makes it seem at a glance that there's something new in the starboard API. This is already happening with #1225 going in, but I think ideally there would be a separate directory for functions used in both starboard and cobalt that aren't part of the starboard API like this. Definitely open to a discussion about this though

Reply:
IMO it makes sense to create another top level //common or //helpers directory that both starboard and cobalt use

Reply:
The `starboard/common` is the current place to have code shared between Cobalt and Starboard. 
If we have a problem with that we can have a separate discussion and not affect this PR. The naming isn't perfect, but that's our current approach.

Reply:
I think we should have a discussion about it, but I'm fine with this PR adding to it as is